File: this_week/piggy/merge_sorted_array.c
#include<stdio.h>

#define TOTAL 10

void merge(int*, int, int*, int);

int main(void) {
    int nums1[TOTAL] = {1, 3, 6, 7};
    int nums2[3] = {2, 4, 9};

    merge(nums1, 4, nums2, 3);

    for (int i = 0; i < 7; i++)
        printf("%d ", nums1[i]);

    printf("\n");

    return 0;
}

void merge(int* nums1, int m, int* nums2, int n) {
    int i = m + n - 1;
    int j = m - 1;
    int k = n - 1;

    for (; i >= 0; i--)
        nums1[i] = (nums1[j] >= nums2[k] ? nums1[j--] : nums2[k--]);
}

File: 02/pierpont/Merge Sorted Array.java

public class number_five {
	public static void main(String[] args) {
		int[] nums1 = {1,3,6,7,9,0,0,0,0,0,0};
		int[] nums2 = {2,4,7,8,0,0};
		merge(nums1,5,nums2,4);
		for(int a = 0;a < 9;a++) {
			System.out.print(nums1[a]);
		}
	}
	
    public static void merge(int[] nums1, int m, int[] nums2, int n) {
    	m--;
    	n--;
        for(int i = m + n + 1;i >= 0; i--) {
        	if (m < 0||n < 0) {
        		if (m < 0) {
        			nums1[i] = nums2[n];
        			n--;
        		} else if (n < 0) {
        			nums1[i] = nums1[m];
        			m--;
        		}
        	}  else {
        		if (nums1[m] > nums2[n]) {
            		nums1[i] = nums1[m];
            		m--;
            	} else if (nums1[m] < nums2[n] ||
            			nums1[m] == nums2[n]) {
            		nums1[i] = nums2[n];
            		n--;
            	}
           }
      }
   }
}